tests/testthat/test-coxph.R

```diff
@@ -1,114 +1,113 @@
-if (requiet("testthat") &&
-  requiet("insight") &&
-  requiet("survival")) {
-  lung <- subset(lung, subset = ph.ecog %in% 0:2)
-  lung$sex <- factor(lung$sex, labels = c("male", "female"))
-  lung$ph.ecog <- factor(lung$ph.ecog, labels = c("good", "ok", "limited"))
-
-  m1 <- coxph(Surv(time, status) ~ sex + age + ph.ecog, data = lung)
-
-  test_that("model_info", {
-    expect_true(model_info(m1)$is_logit)
-    expect_false(model_info(m1)$is_linear)
-  })
+requiet("testthat")
+requiet("insight")
+requiet("survival")
```

Comment:
Why did you remove the if-statement here?

Reply:
The canonical way to skip tests in `testthat` (recommended by the devs) is to use `testthat::skip_if_not_installed()`. This is a matter of taste, but I feel that it is much cleaner to use skips than to wrap everything in big loops, which increase clutter and indentation needlessly. What I did here was to include a call to `testthat::skip_if_not_installed()` in the `requiet()` function. That way, if we try to load a library that is not installed, the test will be skipped automatically by `testthat`. A further benefit of this approach is that by explicitly skipping test instead, we get a nice count of the number of skipped test in the `testthat` output. With the loops, those tests are skipped silently and not reported anywhere.

Reply:
FYI, if `skip_*` is at the top of the script, then everything is skipped. You can also include a `skip_ci`, `skip_cran`, or `skip_if_not_installed` **inside** a `test_that({})` call, in which case just that test will be skipped.